static/src/javascripts/projects/commercial/modules/braze.js

```diff
@@ -0,0 +1,62 @@
+// @flow
+
+import { onIabConsentNotification } from '@guardian/consent-management-platform';
+import config from 'lib/config';
+import { getUserFromApi } from '../../common/modules/identity/api';
+
+
+const getBrazeUuid = (): Promise<string> =>
+    new Promise(resolve => {
+        getUserFromApi(user => {
+            if (user && user.privateFields && user.privateFields.brazeUuid){
+                resolve(user.privateFields.brazeUuid)
+            }
+        })
+    })
+
+const hasRequiredConsents = (): Promise<void> =>
+    new Promise(resolve => {
+        onIabConsentNotification(state => {
+            if (state[1] && state[2] && state[3] && state[4] && state[5]) {
+                resolve()
+            }
+        })
+    })
+
+export const init = (): Promise<any> => {
+    const brazeSwitch = config.get('switches.brazeSwitch');
+    if (!brazeSwitch) return Promise.resolve();
+
+    const apiKey = config.get('page.brazeApiKey');
+    if (!apiKey) return Promise.reject(new Error('Braze API key not set.'));
+
+    const dependencies = [getBrazeUuid(),hasRequiredConsents()]
```

Comment:
The Braze UUID comes from a request to `idapi/user/me` from the browser. Requests to that endpoint are memoized, so the call here doesn't actually result in any additional API requests being made, since it seems we're already calling that endpoint on every page. Perhaps we could optimize that, but it feels out of scope for this PR.